src/pages/Shelter/components/ShareButton/ShareButton.tsx

```diff
@@ -0,0 +1,70 @@
+import { useRef } from 'react';
+import { Share2 } from 'lucide-react';
+
+import { Button } from '@/components/ui/button';
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog';
+
+import { Input } from '@/components/ui/input';
+import { Label } from '@/components/ui/label';
+
+export function ShareButton() {
+  const urlRef = useRef(null);
+
+  const copyToClipboard = () => {
+    urlRef.current.select();
+    document.execCommand('copy');
+  };
+
+  return (
+    <Dialog>
+      <DialogTrigger asChild>
+        <Button variant="ghost">
+          <Share2 />
+        </Button>
+      </DialogTrigger>
+      <DialogContent className="sm:max-w-[450px]">
+        <DialogHeader>
+          <DialogTitle>Compartilhar</DialogTitle>
+        </DialogHeader>
+        <div className="grid gap-4 py-4">
+          <div className="grid grid-cols-3 items-center gap-4">
+            <Button variant="outline" onClick={copyToClipboard}>
+              <Label htmlFor="url" className="text-right">
+                Copiar link
+              </Label>
+            </Button>
+            <Input
+              id="url"
+              defaultValue={window.location.href}
+              className="col-span-2"
+              readOnly={true}
+              ref={urlRef}
+            />
+          </div>
+          <div className="grid grid-cols-3 items-center gap-4">
+            <a
+              href={`https://wa.me/?text=${window.location.href}`}
+              target="_blank"
+            >
+              <Button>WhatsApp</Button>
+            </a>
```

Comment:
Você pode usar a prop _asChild_ e renderizar somente o elemento âncora.

```suggestion
    <Button asChild>
        <a href={`https://wa.me/?text=${window.location.href}`} target="_blank">
          WhatsApp
        </a>
    </Button>
```

Reply:
Semanticamente falando, eu imagino que renderizar a link dentro dentro de um button fere esse conceito. 

Uma sugestão, seria colocar os ícones dessas redes socias, os usuários geralmente já estão acostumados com essa associação.

Reply:
ok!

Reply:
O elemento button não vai estar na DOM.

O conceito da prop _asChild_ é passar as propriedades do **elemento pai** para o **elemento filho**, e dessa maneira você elimina a renderização do **elemento pai**.

Reply:
![image](https://github.com/SOS-RS/frontend/assets/65927539/7db499b3-79d9-4d05-9af8-ecd04ec20ad1)

e isso?

Reply:
Isso mesmo, show de bola!